### ISSUE TITLE
Belongs to relationships are not loading as expected

### ISSUE THREAD

**Issue by roomman** [mikkopaderes/ember-cloud-firestore-adapter#268]
Possibly not a bug but I am expecting relationships to be loaded asynchronously and they are not. For example, I have a project with Bookings that belong to one of my Customers. When loading a Booking into an edit form, I have to make a separate request to fetch the Customer, unless that record happens to have been loaded elsewhere.

Is this an expected behaviour? Is there a recommended way of loading in relationships?

**Comment by charlesfries** [comment]
> I have to make a separate request to fetch the Customer, unless that record happens to have been loaded elsewhere.

Unless I am misunderstanding, this is how EmberData async relationships are actually supposed to work.

In JS:

```js

const booking = await this.store.findRecord('booking', '');
const customer = booking.customer; // initially null, but makes request
const awaitedCustomer = await booking.customer; // Customer
```

In templates:

```hbs
{{#if booking.customer.isSettled}}
  {{booking.customer.id}}
{{else}}
  Loading
{{/if}}
```

**Comment by mikkopaderes** [comment]
@charlesfries is correct.

You have to preload `Booking` beforehand as per nature of [async relationships](https://guides.emberjs.com/release/models/relationships/#toc_relationships-as-promises). I don't provide an API to prettify any side-loading of relationships as Firestore JS SDK doesn't provide it themselves as far as I know, so you're really left to do it on your own. Something like:

```javascript
async model(params) {
  const project = await this.store.findRecord('project', params.project_id);

  // Preload Booking (syntax below may cause some lint errors)
  await project.booking;

  return project;
}
```

**Comment by roomman** [comment]
> ```javascript
> 
> async model(params) {
> 
>   const project = await this.store.findRecord('project', params.project_id);
> 
> 
> 
>   // Preload Booking (syntax below may cause some lint errors)
> 
>   await project.booking;
> 
> 
> 
>   return project;
> 
> }
> 
> ```

Excellent, thanks, this is what I am doing.

**Comment by roomman** [comment]
> const customer = booking.customer; // initially null, but makes request, unless record already loaded

Thanks @charlesfries. If I use `{{booking.customer.name}}` in a template (i.e. I'm not making a request for the customer in the route or backing class, and leave Ember to do its think) nothing happens, like the request isn't being made. When using the default JSON:API adapter I feel like I wouldn't need the manual requests and binding to the template would be enough the fire the request asynchronously. Or am I making that up? 🤣

**Comment by mikkopaderes** [comment]
> I wouldn't need the manual requests and binding to the template would be enough the fire the request asynchronously

That should be the same for this adapter unless there's a bug. 😳

**Comment by roomman** [comment]
Thanks @mikkopaderes, in which case I will attempt a minimal reproduction and post a link here. 👍🏻

**Comment by roomman** [comment]
@mikkopaderes, I'm super sorry about the delay in getting this reproduction together: https://github.com/roomman/reproduction-ecfa-realtime-relationships-bug

Clone this project and, after `npm install`, please run `ember s` and `firebase emulators:start --import=saved-data`.

The reproduction is very simple: an index route that loads two Booking models from the Firestore and renders each in a list. 

Each booking has a belongsTo relationship to the same Customer in Firestore. When the relationship `isRealtime` the Customer model is only available to the first Booking in the list.
<img width="321" alt="Screenshot 2023-06-22 at 17 46 39" src="https://github.com/mikkopaderes/ember-cloud-firestore-adapter/assets/10168451/c1b3e892-1fc5-4a24-acf2-fcff8949cf16">


Without `isRealtime` the Customer model is available to both Bookings.
<img width="226" alt="Screenshot 2023-06-22 at 17 45 55" src="https://github.com/mikkopaderes/ember-cloud-firestore-adapter/assets/10168451/15145d2d-b0a5-4681-a34d-1e8b20c4cb28">

**Comment by mikkopaderes** [comment]
Thanks for taking the time to provide a replication. This should be fixed in the recent `v3.2.1` release.
